template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -468,9 +468,27 @@ public ApiClusterTemplate getTemplate() {
     }
 
     public void addHosts(Map<String, List<Map<String, String>>> hostGroupMappings) {
-        hostGroupMappings.forEach((hostGroup, hostAttributes) -> hostAttributes.forEach(
-                attr -> cmTemplate.getInstantiator().addHostsItem(new ApiClusterTemplateHostInfo().hostName(attr.get("fqdn")).hostTemplateRefName(hostGroup))
-        ));
+        ApiClusterTemplateInstantiator instantiator = cmTemplate.getInstantiator();
+        Map<String, List<String>> roleRefNameMap = null;
```

Comment:
Nit: Why not call this directly `finalRoleRefNameMap`? I do not see the justification for having `roleRefNameMap` as it is only used to initialize `finalRoleRefNameMap`.

Reply:
That can be done by re-arranging the code.

Here I can not perform the initialization without this check `hosts != null && !hosts.isEmpty()`. IF I have to then I need to move the null check to the fetch function. I can not actually clear the hosts' entry in a function named fetch. So need to find a new name for the function as well.

The code was structured the other way, but only to realize I can not use the reference inside a lambda unless it is a final.

Reply:
Ah OK, got it. Makes sense. 😃